src/esperanto/NewsletterBundle/EventListener/NewsletterBuilderSubscriber.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace esperanto\NewsletterBundle\EventListener;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\DependencyInjection\Loader;
+use Symfony\Component\HttpKernel\DependencyInjection\Extension;
+use esperanto\AdminBundle\DependencyInjection\Configuration;
+use Symfony\Component\Config\Definition\Processor;
+
+
+class NewsletterBuilderSubscriber extends Controller implements EventSubscriberInterface
+{
+    /**
+     * @var Container
+     */
+    protected $container;
+    private $subscriber;
+
+    public function __construct(Container $container, $subscriber)
+    {
+        $this->subscriber = $subscriber;
+        $this->container = $container;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+            'esperanto_newsletter.subscriber.pre_create' => array('onPreCreate', 0),
+            'esperanto_newsletter.subscriber.post_create' => array('onPostCreate', 0),
+        );
+    }
+
+    public function onPreCreate(GenericEvent $event)
+    {
+        $subscriber = $event->getSubject();
+
+        //Erstellungszeitpunkt in DB speichern
+        $subscriber->setCreated(new \DateTime());
+
+        //Token
+        $tokenId = time();
+        $tokenId = $tokenId.$subscriber->getEmail();
+        $csrfToken = $this->get('security.csrf.token_manager')->getToken($tokenId)->getValue();
```

Comment:
its not wrong to use csrf token. It should be secure enough. But there is also a service called "security.secure_random" witch also provide token without any csrf context. Better use this.